components/dependency_injection/autowiring.rst

```diff
@@ -0,0 +1,397 @@
+.. index::
+    single: DependencyInjection; Autowiring
+
+Defining Services Dependencies Automatically
+============================================
+
+Autowiring allows to register services in the container with minimal configuration.
+It is useful in the field of `Rapid Application Development`_, when designing prototypes
+in early stages of large projects. It makes it easy to register a service graph
+and eases refactoring.
```

Comment:
I think we should mention what autowiring is here. E.g. start the paragraph with "Autowiring automatically resolves the service dependencies, based on the typehint of the constructor."